### PR TITLE
Improve app handling from stack and registry

### PR DESCRIPTION
**Pull request by CPatchane**

### CODE REVIEW

**Comment by CPatchane** [comment]
[(1/5) [STO-8] ETU je peux accéder aux informations de l'application (lecture)](https://trello.com/c/J4LqVadR/632-1-5-sto-8-etu-je-peux-acc%C3%A9der-aux-informations-de-lapplication-lecture)

**Comment by Narkfr** [comment]
a lot of refact, lgtm !

File: src/ducks/apps/currentAppVersion/index.js [hunk 1]
```diff
@@ -53,7 +53,22 @@ export function fetchLastAppVersion (appSlug, channel = 'stable') {
...
+      const versionFromRegistry = version.version
+      const manifest = version.manifest
+      const screensLinks = manifest.screenshots && manifest.screenshots.map(name => {
+        const fileName = name.replace(/^.*[\\/]/, '')
+        return `${cozy.client._url}/registry/${manifest.slug}/${versionFromRegistry}/screenshots/${fileName}`
+      })
+      const iconLink = `${cozy.client._url}/registry/${manifest.slug}/${versionFromRegistry}/icon`
+      return dispatch({
+        type: FETCH_APP_VERSION_SUCCESS,
+        version: Object.assign({}, version, {
```
Comment (gregorylegarec) [inline src/ducks/apps/currentAppVersion/index.js]: Why don't use the spread operator here ?
```js
return dispatch({
  // ...
  version: {
    ...version,
   manifest: {
     ...manifest,
     screenshots: {
       // ...
     }
  }
})
```

**Comment by gregorylegarec** [comment]
Thanks @CPatchane ! 🔍

**Pull request merged**